File: docs/manual/guide/cluster/code/docs/home/persistence/BlogService.java
package docs.home.persistence;

import com.lightbend.lagom.javadsl.api.*;
import com.lightbend.lagom.javadsl.api.transport.*;

import static com.lightbend.lagom.javadsl.api.Service.*;

public interface BlogService extends Service {

  ServiceCall<BlogCommand.AddPost, String> addPost(String id);

  @Override
  default Descriptor descriptor() {
    return named("/blogservice").withCalls(
      restCall(Method.POST, "/blogs/:id", this::addPost)
    );
  }
}

File: docs/manual/scala/guide/cluster/code/docs/home/scaladsl/persistence/BlogService.scala
/*
 * Copyright (C) 2016-2019 Lightbend Inc. <https://www.lightbend.com>
 */

package docs.home.scaladsl.persistence

import scala.concurrent.Future
import akka.NotUsed
import akka.stream.scaladsl.Source
import com.lightbend.lagom.scaladsl.api.Service
import com.lightbend.lagom.scaladsl.api.ServiceCall

trait BlogService extends Service {
  def getPostSummaries(): ServiceCall[NotUsed, Source[PostSummary, _]]

  override def descriptor = ???
}
